src/flow/mod.rs

```diff
@@ -1191,14 +1191,28 @@ impl<'a> Migration<'a> {
         // etc.
         // println!("{}", mainline);
 
-        let mut uninformed_domain_nodes = domain_nodes.clone();
-        let deps = migrate::transactions::analyze_graph(
+        migrate::transactions::analyze_graph(
             &mainline.ingredients,
             mainline.source,
-            domain_nodes,
+            domain_new_nodes,
+            &mut mainline.deps,
```

Comment:
Any particular reason you chose to do the merge inside `analyze_graph` instead of returning `new` from there and doing the merge here (which would save you an argument to `analyze_graph`)?

Reply:
Not particularly, although now that I think about it, I should probably rename it to `update_deps` or something.

Reply:
I'd say make it `analyze_changes`, have it return `new`, and update in caller. Improves modularity.